src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -139,24 +139,22 @@ int goto_instrument_parse_optionst::doit()
     instrument_goto_program();
 
     {
-      bool unwind=cmdline.isset("unwind");
-      bool unwindset=cmdline.isset("unwindset");
-      bool unwindset_file=cmdline.isset("unwindset-file");
+      bool unwind_given=cmdline.isset("unwind");
+      bool unwindset_given=cmdline.isset("unwindset");
+      bool unwindset_file_given=cmdline.isset("unwindset-file");
 
-      if(unwindset && unwindset_file)
+      if(unwindset_given && unwindset_file_given)
         throw "only one of --unwindset and --unwindset-file supported at a "
               "time";
 
-      if(unwind || unwindset || unwindset_file)
+      if(unwind_given || unwindset_given || unwindset_file_given)
       {
-        int k=-1;
+        unwindsett unwindset;
 
-        if(unwind)
-          k=std::stoi(cmdline.get_value("unwind"));
+        if(unwind_given)
+          unwindset.parse_unwind(cmdline.get_value("unwind"));
 
-        unwind_sett unwind_set;
-
-        if(unwindset_file)
+        if(unwindset_file_given)
```

Comment:
Would it be possible to move all this code to `unwindset.cpp` as well? `*parse_options` tends to be way too big anyway...

Reply:
that adds to the complexity of all users, say jbmc, cbmc; mildly reluctant

Reply:
Just to be clear: I mainly wanted the code to move, not necessarily add that to the external user interface. Factoring out what belongs together would have seemed like a good idea?!

But also: It might actually be desirable to open up this additional option to all users?

Reply:
ok, done